lib/opal/nodes/literal.rb

```diff
@@ -138,11 +138,11 @@ def extract_flags_and_value
         *values, flags_sexp = *children
         self.flags = flags_sexp.children.map(&:to_s)
 
-        case values.length
-        when 0
+        l = values.length
+        if l == 0
           # empty regexp, we can process it inline
           self.value = s(:str, '')
-        when 1
+        elsif l == 1 && put_inline?(values)
```

Comment:
can you change to `put_inline?` method name to something that better reflects the check on having a single line? (for example `single_line?`)

alternatively it could be the only thing in the `elsif` condition:

```rb
if values.length == 0
  # …
elsif put_inline?(values)
  # …

…

def put_inline?(values)
  return false if values.length > 1
  # …
end
```